### DISCUSSION TITLE
Synchronous findBody

### DISCUSSION THREAD

**Discussion by Toby222** [shakiba/planck.js#170] [Q&A] [answered]
Taking from the example in testbed, it finds a clicked body with the method `findBody`, passing any found body to a callback.

Is there some way to make a findBody method that works synchronously and returns the found body instead?

**Comment by zOadT** [comment, marked as answer]
Maybe we are talking about different functions here, but the `findBody` function in `testbed/index.js` returns the body synchonously, see line 278 (but uses a function passing all found fixtures to a callback)
Does this function solve your issue or did I misunderstand your question? 😅

**Reply by shakiba** [reply]
Yup, it is sync. It is a callback just so that you can stop query by returning false.

**Reply by Toby222** [reply]
Yea, that answers that and makes me wonder what I was even doing z-z
